Assets/Scripts/Models/Job/JobManager.cs

```diff
@@ -13,26 +13,29 @@
 
 public class JobManager
 {
-    private HashSet<Job> jobQueue;
+    private Dictionary<JobCategory, HashSet<Job>> jobQueue;
+
+    private CharacterJobPriority[] characterPriorityLevels = (CharacterJobPriority[])Enum.GetValues(typeof(CharacterJobPriority));
+
+    private JobCategory[] categories;
 
     public JobManager()
     {
-        jobQueue = new HashSet<Job>();
+        jobQueue = new Dictionary<JobCategory, HashSet<Job>>();
+        categories = PrototypeManager.JobCategory.Values.ToArray();
+        foreach (JobCategory category in categories)
+        {
+            jobQueue[category] = new HashSet<Job>();
+        }
     }
 
-    public event Action<Job> OnJobCreated;
+    public delegate void JobChanged(Job job);
 
-    public bool IsEmpty()
-    {
-        return jobQueue.Count == 0;
-    }
+    public event JobChanged OnJobCreated;
 
-    // Returns the job count in the queue.
-    // (Necessary, since jobQueue is private).
-    public int GetCount()
-    {
-        return jobQueue.Count;
-    }
+    public event JobChanged OnJobModified;
+
+    public event JobChanged OnJobRemoved;
```

Comment:
These should be in the format of JobChanged, without the "On". The "On<action>" format is what we would use for the functions that get passed to these (the observer I suppose).